### PR TITLE
Support UMD usage of QSF tools

### PR DESCRIPTION
**Pull request by nmdefries**
### Description
Generalize QSF tools to work for UMD as well.

Add UMD static files (item descriptions, etc). Define utils function to adjust path to static files based on survey version (UMD vs CMU). Percolate survey version field through functions correctly.

### Changelog
- `generate-codebook.R`
    - Create path for static files to use either UMD or CMU `static` subdirectory
    - Rearrange logic that fetches choices and answers from the QSF to work better with matrix items. Matrix export field names can be specified in the QSF (UMD does this), so we need to fetch those from `ChoiceDataExportTags` and only generate auto-incrementing matrix field names when not available.
    - Instead of overwriting any existing answer choices for a question with carryforward choices, combine the two.
- `qsf-utils.R`: `patch_item_names` (e.g. "B13 " -> "B13") now uses `static/<survey version>/item_rename_map.csv` instead of being hardcoded
- `CMU/item_shortquestion_map.csv`: add descriptions for `V17_month` and `V17_year`
- new static files:
    - `CMU/item_drop.csv`
    - `CMU/item_rename_map.csv`
    - `UMD/item_drop.csv`
    - `UMD/item_rename_map.csv`
    - `UMD/item_rename_map.csv`
    - `UMD/item_shortquestion_map.csv`
    - `UMD/static_microdata_fields.csv`

### CODE REVIEW

File: facebook/qsf-tools/generate-codebook.R [hunk 1]
```diff
@@ -353,16 +411,27 @@ add_qdf_to_codebook <- function(qdf,
...
+      !(variable == "w12_treatment" & wave != 12.5) # experimental arm field is only available for wave == 12.5
     )
-
+  } else if (survey_version == "UMD") {
+    codebook <- filter(
+      codebook,
+      !(variable == "module" & wave < 11), # module field is only available for wave >= 11
+      !(variable %in% c("wave", "UserLanguage", "fips") & wave < 4), # wave, UserLangauge, and fips fields are only available for wave >= 4
+      !(variable == "w12_treatment" & wave != 12.5) # experimental arm field is only available for wave == 12.5
+    )
```
Comment (nmdefries) [inline facebook/qsf-tools/generate-codebook.R:433]: Zheng, you should modify this bit for UMD. It drops those non-item fields that are included in the microdata (`StartDate`, `EndDate`, `wave`, etc) depending on which wave they first appeared in.

File: facebook/qsf-tools/static/UMD/static_microdata_fields.csv [hunk 2]
```diff
@@ -0,0 +1,9 @@
+variable,replaces,description,question,matrix_subquestion,type,response_option_randomization
```
[the remark this answers is not in the document]
Comment (nmdefries) [reply facebook/qsf-tools/static/UMD/static_microdata_fields.csv:1]: @leonardzh tagging you on here so you can follow the discussion. Let me know what you want to do about this static file and the code snippet above. I can throw in those changes whenever they're ready.

**Comment by capnrefsmmat** [comment]
Would it make sense to review/approve this once it runs and UMD has checked all the output? Since I'm not familiar with UMD's QSFs, I don't really have a way to tell if this is correct.

**Comment by nmdefries** [comment]
@WichadaLMK Let me know when you're done with UMD's `item_replacment_map.csv` and I can include it here.

File: facebook/qsf-tools/static/UMD/static_microdata_fields.csv (hunk 2 above)
[the remark this answers is not in the document]
Comment (nmdefries) [reply facebook/qsf-tools/static/UMD/static_microdata_fields.csv]: These match UMD.

File: facebook/qsf-tools/generate-codebook.R (hunk 1 above)
Comment (nmdefries) [reply facebook/qsf-tools/generate-codebook.R]: UserLanguage filter removed.

**Pull request merged**